[338] Previewing examples with contextual HTML (#353)
github/view_component: README.md

This example will make the previews available from <http://localhost:3000/previews>.

#### Adding contextual HTML for previews

> Given the preview component lives in `test/components/previews/cell_component_preview.rb`, template files can be defined at `test/components/previews/cell_component_preview/`.

Reviewer: ```suggestion
Given a preview component in `test/components/previews/cell_component_preview.rb`, template files can be defined in `test/components/previews/cell_component_preview/`.
```

`test/components/previews/cell_component_preview.rb`
```ruby
class CellComponentPreview < ViewComponent::Preview

>   def default

Reviewer: Rails controllers allow templates to be rendered without defining a method in the controller. Might we consider supporting similar behavior, not requiring `#default` to be defined in this case where a template is also provided?

Author: That will involve changing the concept of an "example" in the Preview object. Currently one example is just one method in the preview object. The example might call render in its body (initial previewing approach) or have an empty body and use a template (new approach).

But we definitely can support this. We would just need to update the preview object.

Author: Well I went a bit different about this, just with the intention to keep the implementation simple. I introduced a specific method to render a preview with a template and make the preview object have all the examples defined in there as methods, as we currently do.

This example will make the previews available from <http://localhost:3000/previews>.

> #### Adding contextual HTML for previews

Reviewer: ```suggestion
#### Adding contextual HTML for previews

```

> #### Adding contextual HTML for previews

Reviewer: ```suggestion
#### Preview templates
```

> Given a preview component`test/components/previews/cell_component_preview.rb`, template files can be defined at `test/components/previews/cell_component_preview/`:

Reviewer: ```suggestion
Given a preview `test/components/previews/cell_component_preview.rb`, template files can be defined at `test/components/previews/cell_component_preview/`:
```

> Given the preview component lives in `test/components/previews/cell_component_preview.rb`, template files can be defined at `test/components/previews/cell_component_preview/`.

Reviewer: ```suggestion
Given a preview component`test/components/previews/cell_component_preview.rb`, template files can be defined at `test/components/previews/cell_component_preview/`:
```

`test/components/previews/cell_component_preview.rb`
```ruby
class CellComponentPreview < ViewComponent::Preview
  def default
    render_with_template
  end
end
```

`test/components/previews/cell_component_preview/default.html.erb`
```erb
<table class="table">
  <tbody>
    <tr>
      <%= render CellComponent.new %>
    </tr>
  </tbody>
</div>
```

> If you'd rather use a different location for your custom preview templates, you can pass a path in the `template` argument

Reviewer: ```suggestion
To use a different location for preview templates, pass the `template` argument:
```

> (the path should be relative to your `config.view_component.preview_path`):

Reviewer: ```suggestion
(the path should be relative to `config.view_component.preview_path`):
```

`test/components/previews/cell_component_preview.rb`
```ruby
class CellComponentPreview < ViewComponent::Preview
  def default
    render_with_template(template: 'custom_cell_component_preview/my_preview_template')
  end
end
```

> To use dynamic values from the params you can use the `locals` argument:

Reviewer: ```suggestion
Values from `params` can be accessed through `locals`:
```
